pyiceberg/table/__init__.py

```diff
@@ -192,6 +195,14 @@ class TableProperties:
     WRITE_PARTITION_SUMMARY_LIMIT = "write.summary.partition-limit"
     WRITE_PARTITION_SUMMARY_LIMIT_DEFAULT = 0
 
+    WRITE_LOCATION_PROVIDER_IMPL = "write.location-provider.impl"
```

Comment:
Though the [docs](https://iceberg.apache.org/docs/1.7.0/configuration/#write-properties) say that the default is `null`, having a constant for this being `None` felt unnecessary